### DISCUSSION TITLE
CommandProcess ClearOutbox will send a message regardless of if its been sent before

### DISCUSSION THREAD

**Discussion by preardon** [BrighterCommand/Brighter#1600] [Ideas]
@iancooper I was wondering if this was intended

if you run ClearOutbox with an Id of a message there are no checks at the Command Processor level to see if it has already been dispatched, I understand that this may be on Purpose (Re dispatching messages), I was just wondering if you think it would be a good idea to add a bool to the Clear Outbox Method to turn on a already dispatched check?

I understand that systems need to be idempotent and be able to consume duplicate events, and I can only see this being of benefit when someone Deposits into the outbox and then for some reason there is a wait between that and Clearing the outbox where the Outbox sweeper could get in the middle. not urgent just wanted your thoughts on the matter.

**Comment by iancooper** [comment]
I guess the trade-off is on the cost off querying the Outbox. I don't have an objection to adding a deDuplicate bool to Clear and defaulting it to false, and then letting folks choose to check if they want to.

**Reply by preardon** [reply]
That was my thoughts, don't enforce it but give the option, again, not a big issue but a nice to have
